### ISSUE TITLE
a question about psptool/fet.py:ZEN_GENERATION_IDS

### ISSUE THREAD

**Issue by hanetzer** [PSPReverse/PSPTool#46]
Is there any difference in psp-space between a Zen 1 and a Zen+ PSP?
I'm attempting to use the tool to dissect a vendor bios for psp parts to be
incorporated into a coreboot.rom image, in a manner similar to how intel
coreboot images incorporate the intel flash descriptor, management engine
firmware, and so forth into the final product, as my current (2700x/Zen+)
and future target (5950x/Zen 3) cpus do not have a released firmware image
in the coreboot amd_blobs repository to use.

**Comment by cwerling** [comment]
As far as we know, there are only subtle to no differences in the PSP space between Zen 1 and Zen+.

I didn't quite understand from your message what's your status quo and where you want to go. Are you saying that you have a working coreboot setup on a Zen(+) CPU and want to upgrade to Zen 3? 
I'm not a coreboot expert, but as far as I understood the compatibility state of coreboot, none of the Zen CPUs are supported yet.

In the server space, there was an announcement recently about open AGESA components: https://www.phoronix.com/news/AMD-openSIL-Open-Source

**Comment by hanetzer** [comment]
No, I have currently no working coreboot for non-chromeos ryzen devices.
I just wish to be able to tell from psptool output, which directory would be
used on any given cpu.

**Comment by cwerling** [comment]
Well, some of the directories can be attributed to the Zen version with PSPTool (see screenshot of an ASRock B350), but not all of them.

<img width="1246" alt="Bildschirmfoto 2023-07-26 um 11 30 05" src="https://github.com/PSPReverse/PSPTool/assets/8776698/b11d17cc-cfe4-41a1-a5a2-dddb92f4fc51">

**Comment by cwerling** [comment]
I don't think so. We got them from dumping and reverse-engineering the On-Chip Boot Loaders of different Zen generations. On Zen 1, this can be done with the known ROM buffer overflow, on Zen 2 and Zen 3 the glitch attack allowed us to dump them.

**Comment by hanetzer** [comment]
Gotcha. So, with a 'simple setup' with commonly held hardware hacking tools, the best way to determine what psp modules load on a given bios & cpu combination would be to psptrace it?
